### DISCUSSION TITLE
SUPG tau stabilization parameter for temperature

### DISCUSSION THREAD

**Discussion by smpark7** [idaholab/moose#17185] [Q&A Modules: Navier-Stokes] [answered]
I've been using the INSAD kernels with SUPG and PSPG to model molten salt flow and temperature advection-diffusion in a lid-driven cavity with a volumetric heat source (fission) and sink. The tau parameter for temperature advection-diffusion in ```INSADStabilized3Eqn.C``` has an additional term in transient simulations that depends on the timestep size (4/dt**2).

For non-large timesteps in advection-dominated flow, this timestep term effectively eliminates the artificial diffusion from SUPG on "non-physical" boundary points like the top left corner of the lid-driven cavity where the velocity value is zero. This causes the temperature value on the corner to plunge and creates spurious oscillations in the temperature along the boundary when using natural BCs.

I looked through the MOOSE INS Overview paper and tried looking at other resources but I couldn't find any references for the derivation of this term. Is this term in tau necessary for transient simulations?

**Comment by GiudGiud** [comment]
Tagging @pbalest and @snschune  as they will know more on what reference was followed for this implementation. 
Is stabilization required for your use case?

**Reply by smpark7** [reply]
Yes the prandtl number of the lid-driven cavity with molten salt is ~300k so it produces worse numerical oscillations in the temperature without stabilization.

I also tried removing the timestep-dependent term in the tau parameter before running with SUPG stabilization. At steady state, the temperature values along the vertical and horizontal lines passing through the middle of the cavity are close to literature-reported values. The "non-physical" temperature value observed on the top-left corner is steady and does not induce oscillations in other parts of the domain like what I described in my initial post above.

**Reply by GiudGiud** [reply]
Thanks for the additional information. 
Would you consider using our new finite volume capabilities? It should be able to remove unphysical corner values which can be simply caused by difficulties in applying boundary conditions there. 

There are examples of lid-driven cavities in modules/navier_stokes/test/tests/finite_volume/ins/lid-driven

**Reply by smpark7** [reply]
Yes I'll definitely look into INSFV for future work!

**Reply by lindsayad** [reply]
@jwpeterson I believe I added that transient contribution to the tau parameter based off information in one of your fluids textbooks which I liked to steal and unfortunately can no longer steal 😆 I don't suppose you know the reference?

**Reply by lindsayad** [reply]
We want to emphasize our FV capability but I still want to ensure that our FE capabilities are as good as they can be

**Reply by smpark7** [reply]
Thanks for checking in on this thread. May I ask what's the current status on coupling and other supporting features in the NS FV system? I've seen the conference papers on Griffin & Pronghorn which showcased tight coupling through the multiapp system (please correct me if I'm wrong). Is direct coupling to FE variables a work-in-progress?

**Reply by lindsayad** [reply]
We have coupled FV variables to FE physics, but not the other way around yet. @csdechant was working on FE variables to FV physics in https://github.com/idaholab/moose/pull/18394 but we seem to have stalled a little.

I have been focused on the fluid flow physics development for FV, but others like @GiudGiud have applied it in more multiphysics contexts, with neutronics for instance. For neutronics coupling I believe that has been done with multiapps. But energy transport and passive scalar transport can certainly be solved in a single monolithic system (using FV variables for all)

**Comment by jwpeterson** [comment, marked as answer]
> I don't suppose you know the reference?

I'm not sure what the original reference is, but my understanding is that this "sum of reciprocals" formula became a pseudo-standard approach in the early literature. See for example Eq. (18) and the discussion leading up to it in Tezduyar & Osawa, CMAME 190 (2000), p. 411-430 (attached). 

The main idea is that in the limit as dt -> 0, tau -> dt/2, which I think relates back to the (even older) Taylor-Galerkin  stabilization method. On the other hand if dt is not small and |u|~0, then tau -> C * h^2 / nu, i.e. the classical diffusive limit for SUPG.  

> Yes the prandtl number of the lid-driven cavity with molten salt is ~300k

The Prandtl number for molten salt is .003 according to [Wikipedia](https://en.wikipedia.org/wiki/Prandtl_number)?

[Tezduyar_CMAME_2000.pdf](https://github.com/idaholab/moose/files/7590577/Tezduyar_CMAME_2000.pdf)

**Reply by smpark7** [reply]
Thank you for the references. I've also since found one other publication by the same author providing that tau formulation ([link](https://www.sciencedirect.com/science/article/abs/pii/0045782592901416)), but without as much derivation.
> The Prandtl number for molten salt is .003 according to Wikipedia?

That appears to be the Prandtl number for molten potassium metal, rather than molten salt.

**Reply by jwpeterson** [reply]
OK, well, a Prandtl number that large would mean that the thermal boundary layer is much, much, much thinner than the momentum boundary layer, so the mesh that you would need to capture the thermal boundary layer accurately would be much finer than that needed for capturing the velocity boundary layer. You might have better luck with a decoupled approach, as I would also expect thermal changes and momentum changes to happen on very different time scales, i.e. the two things would be almost decoupled from one another physically.

**Reply by smpark7** [reply]
I see. Thank you for the insights!

**Reply by doquang** [reply]
@jwpeterson - Do you mean the decoupled approach is the MultiApp in Moose?

**Reply by jwpeterson** [reply]
@doquang Yes, MultiApps would be one way to implement a decoupled approach in MOOSE. This would allow you to e.g. use different time steps and different meshes for the momentum and thermal equations. I don't know much about the details of what you are doing, though, so decoupling the equations is just my initial instinct upon hearing about the high Prandtl number. Keeping the equations coupled would make sense if the Pr was up to O(10) or so.
